lib/src/local_working_copy.rs

```diff
@@ -1021,6 +1021,7 @@ impl TreeState {
     ) -> Result<FsmonitorMatcher, SnapshotError> {
         let (watchman_clock, changed_files) = match fsmonitor_kind {
             None => (None, None),
+            Some(FsmonitorKind::None) => (None, None),
```

Comment:
Perhaps, `UserSettings::fsmonitor_kind()` can be changed to just return `FsmonitorKind` because it doesn't make sense to have two different encodings for `None`.

Alternatively, `"none"` can be handled there.

Reply:
Oh, good idea. Now that I made this change and uploaded it, I'm not sure if it's better to just handle `"none"` in `UserSettings::fsmonitor_kind()` as a one-line change, or do this. Maybe you and Waleed can tie break on which approach is better. :)

Reply:
I slightly prefer `FsmonitorKind::None` (the current approach.)

`UserSettings::fsmonitor_kind()` could be simplified further if the default were configured in `src/config/*.toml`. I don't know if that's doable though. The lib tests might depend on the empty config value.

Reply:
Yeah, I thought about doing that too, but just didn't get around to it. Maybe I'll poke at it later.